### PR TITLE
TST: streamline orbits tests

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #749, progress on #545, #565, #860

- Reconfigures classes in test_orbits.py for better modularity
  - Two basic inheritable classes: general tests and gappy data tests
  - Reduced length of gappy tests - iterating over one week rather than one month
- Improves style and reduces code duplication
  - usage of parametrize
  - `filter_data`functions combined and generalized
  - local functions renamed to avoid confusion with generalized `filter_data`
- Docstrings for test_orbits.py
- Some duplicate tests removed
- Added ignore D200 and D202 to address docstring linting (#860)

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

tested via pytest.

**Test Configuration**:
* Operating system: Mac OS X 10.15.7
* Version number: Python 3.8.2

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by jklenzing** [comment]
Potentially room for improvement.  This section needs discussion for other tests that can be combined, as well as making sure all use cases are still covered.  Some tests I still don't understand, the docstrings reflect this.  I don't particularly see the rationale for testing for longer gaps than two days, so the five-day gap test has been removed.

Run time down to 30 min.

File: pysat/tests/test_orbits.py [hunk 1]
```diff
@@ -405,160 +434,38 @@ def filter_data(inst):
...
-        for j in range(10):
-            self.testInst.orbits.next()
-        assert all(control.data == self.testInst.data)
-
-    def test_repeated_orbit_calls_antisymmetric_multi_multi_day_off_0_UT(self):
-        """ Test successful orbit calls for more days about a time off 0 UT
-        """
-        self.stime -= dt.timedelta(days=1)
-        self.testInst.load(date=self.stime)
-        self.testInst.orbits.next()
```
Comment (jklenzing) [inline pysat/tests/test_orbits.py:486]: This test (and similar) can be combined with an extra level of parametrize, but I'm not sure it adds anything new

**Comment by rstoneback** [comment]
Taking a look now.

File: pysat/tests/test_orbits.py [hunk 2]
```diff
@@ -12,7 +12,25 @@
...
 
+def filter_data(inst, times=None):
+    """Remove data from instrument, simulating gaps in the dataset"""
+
+    if times is None:
+        times = [[dt.datetime(2009, 1, 1, 1), dt.datetime(2009, 1, 1, 2)],
+                 [dt.datetime(2009, 1, 1, 10), dt.datetime(2009, 1, 1, 12)],
+                 [dt.datetime(2009, 1, 1, 22), dt.datetime(2009, 1, 2, 2)],
+                 [dt.datetime(2009, 1, 4), dt.datetime(2009, 1, 6)]
+                 ]
```
Comment (rstoneback) [inline pysat/tests/test_orbits.py:23]: Some specific edge cases are getting lost.
- single orbit gap
- orbit < gap < 2 orbits
- gap across daybreak, gap > 2 orbits, , no extra data for last orbit on 1st, no data in first orbit time on 2nd
- gap > 1 day, data in first orbit time on 15th, no data on 14th
- gap > 1 day, only a little bit of last orbit on 20th and a little bit of first orbit on 25th available.
- orbit < gap < 2 orbits, daybreak check, 
```
    times = [[dt.datetime(2009, 1, 1, 1, 37), dt.datetime(2009, 1, 1, 3, 14)],
              [dt.datetime(2009, 1, 1, 10), dt.datetime(2009, 1, 1, 12)],
              [dt.datetime(2009, 1, 1, 22), dt.datetime(2009, 1, 2, 2)],
              [dt.datetime(2009, 1, 13), dt.datetime(2009, 1, 15)],
              [dt.datetime(2009, 1, 20, 1), dt.datetime(2009, 1, 25, 23)],
              [dt.datetime(2009, 1, 25, 23, 30), dt.datetime(2009, 1, 26, 3)]
```

File: pysat/tests/test_orbits.py [hunk 3]
```diff
@@ -753,8 +650,41 @@ def teardown(self):
         """Runs after every method to clean up previous testing."""
         del self.testInst, self.stime
 
+    def test_repeat_orbit_calls_asym_multi_day_0_UT_really_long_time_gap(self):
+        """Test successful orbit calls over a series of lengthening gaps
+        """
+        self.testInst.load(date=self.stime)
```
Comment (rstoneback) [inline pysat/tests/test_orbits.py:656]: The loaded data and the number of orbits determines which gap we run into. Based on my list of times, we should load this one on the 19th and go for like 45 orbits.

File: pysat/tests/test_orbits.py [hunk 4]
```diff
@@ -753,8 +650,41 @@ def teardown(self):
...
+        assert all(control.data == self.testInst.data)
+
+    def test_repeat_orbit_calls_asym_multi_day_0_UT_multiple_time_gaps(self):
+        """Test that orbits are selected at same cutoffs backward and forward"""
+        self.testInst.load(date=self.stime)
+        self.testInst.orbits.next()
+        control = self.testInst.copy()
+        n_time = []
+        p_time = []
+        for j in range(40):
```
Comment (rstoneback) [inline pysat/tests/test_orbits.py:672]: 20 orbits ok for this one with a load on 2009, 1

Comment (rstoneback) [reply pysat/tests/test_orbits.py:672]: Also would be good to load on the 25th and go for like 20 orbits.

File: pysat/tests/test_orbits.py [hunk 5]
```diff
@@ -769,7 +699,10 @@ def teardown(self):
         del self.testInst, self.stime
 
 
-class TestOrbitsGappyData2(TestGeneralOrbitsMLT):
+class TestOrbitsGappyData2(TestOrbitsGappyData):
```
Comment (rstoneback) [inline pysat/tests/test_orbits.py:702]: I think the gappydata2 class came later after a bug was found. I don't think it should be running the TestOrbitsGappyData tests.

This class is checking if the first orbit is missing all data, and the first orbit sample starts on the day. Also has a single orbit gap unrelated to daybreak. I think we just need to do a standard iterate forward 20 orbits starting from 2009, 1, back 40, forward 20 type thing here.

File: pysat/tests/test_orbits.py (hunk 2 above)
Comment (jklenzing) [reply pysat/tests/test_orbits.py:23]: Do we need a month of data here, or can we compress into a week and capture all these use cases?

**Comment by jklenzing** [comment]
Changes since last review:
- Using recommended time gaps with targeted tests
- merged in recent changes from develop
- scrubbed docstrings and usage of return

The tests here take between 21 and 37 minutes on GA (unix and windows)
In develop, it takes 50 to 67 minutes.

File: pysat/tests/test_orbits.py [hunk 6]
```diff
@@ -12,26 +12,48 @@
 import pysat
 
 
+def filter_data(inst, times=None):
+    """Remove data from instrument, simulating gaps in the dataset."""
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:16]:
```suggestion
    """Remove data from instrument, simulating gaps in the data set."""
```

File: pysat/tests/test_orbits.py [hunk 7]
```diff
@@ -12,26 +12,48 @@
...
+
+    if times is None:
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:18]: Recommend making this more robust by using the instrument index start time and adding different deltas to it.

File: pysat/tests/test_orbits.py [hunk 8]
```diff
@@ -12,26 +12,48 @@
...
+    """Tests the user interface for orbits, including error handling."""
+
     def setup(self):
-        """ Set up User Interface unit tests
-        """
+        """Set up User Interface unit tests."""
         self.in_args = ['pysat', 'testing']
         self.in_kwargs = {'clean_level': 'clean', 'update_files': True}
         self.testInst = None
         self.stime = dt.datetime(2009, 1, 1)
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:43]: Use instrument testing time

File: pysat/tests/test_orbits.py [hunk 9]
```diff
@@ -12,26 +12,48 @@
...
+        """Tear down User Interface tests."""
         del self.in_args, self.in_kwargs, self.testInst, self.stime
+        return
 
     def test_orbit_w_bad_kind(self):
-        """ Test orbit failure with bad 'kind' input
-        """
+        """Test orbit failure with bad 'kind' input."""
         self.in_kwargs['orbit_info'] = {'index': 'mlt', 'kind': 'cats'}
         with pytest.raises(ValueError):
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:54]: Add an error message evaluation.

File: pysat/tests/test_orbits.py [hunk 10]
```diff
@@ -43,14 +65,14 @@ def test_orbit_w_bad_kind(self):
...
                                         'kind': 'orbit'})])
     def test_orbit_w_bad_orbit_info(self, info):
-        """ Test orbit failure on iteration with orbit initialization
-        """
+        """Test orbit failure on iteration with orbit initialization."""
         self.in_kwargs['orbit_info'] = info
         self.testInst = pysat.Instrument(*self.in_args, **self.in_kwargs)
         self.testInst.load(date=self.stime)
 
         with pytest.raises(ValueError):
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:73]: Add an error message evaluation

File: pysat/tests/test_orbits.py [hunk 11]
```diff
@@ -61,8 +83,7 @@ def test_orbit_w_bad_orbit_info(self, info):
                                       ({'index': 'magnetic local time',
                                         'kind': 'lt'})])
     def test_orbit_polar_w_missing_orbit_index(self, info):
-        """ Test orbit failure oon iteration with missing orbit index
-        """
+        """Test orbit failure oon iteration with missing orbit index."""
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:86]:
```suggestion
        """Test orbit failure on iteration with missing orbit index."""
```

File: pysat/tests/test_orbits.py [hunk 12]
```diff
@@ -71,49 +92,51 @@ def test_orbit_polar_w_missing_orbit_index(self, info):
         self.testInst.load(date=self.stime)
         with pytest.raises(ValueError):
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:93]: Add an error message evaluation.

File: pysat/tests/test_orbits.py [hunk 13]
```diff
@@ -174,55 +197,57 @@ def test_all_single_orbit_calls_in_day(self):
...
             assert self.testInst.index[-1] == self.etime
+        return
 
     def test_orbit_next_call_no_loaded_data(self):
-        """ Test orbit next call without loading data
-        """
+        """Test orbit next call without loading data."""
         self.testInst.orbits.next()
         assert (self.testInst.index[0] == dt.datetime(2008, 1, 1))
         assert (self.testInst.index[-1] == dt.datetime(2008, 1, 1, 0, 38, 59))
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:206]: Should these times be related back to one of the test class attributes?

Comment (aburrell) [reply pysat/tests/test_orbits.py:206]: Here and elsewhere.

File: pysat/tests/test_orbits.py [hunk 14]
```diff
@@ -316,69 +344,56 @@ def test_eval_repr(self):
...
+            """Local helper function to reduce available data."""
             inst.data = inst[0:20]
 
-        self.testInst.custom_attach(filter_data)
+        self.testInst.custom_attach(truncate_data)
         self.testInst.load(date=self.stime)
         self.testInst.orbits.next()
 
-        # a recusion issue has been observed in this area
+        # a recursion issue has been observed in this area
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:373]:
```suggestion
        # A recursion issue has been observed in this area
```

File: pysat/tests/test_orbits.py [hunk 15]
```diff
@@ -316,69 +344,56 @@ def test_eval_repr(self):
...
         # checking for date to limit reintroduction potential
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:374]:
```suggestion
        # Checking for date to limit reintroduction potential
```

File: pysat/tests/test_orbits.py [hunk 16]
```diff
@@ -316,69 +344,56 @@ def test_eval_repr(self):
...
         assert self.testInst.date == self.stime
-
-    def test_less_than_one_orbit_of_data_two_ways(self):
-        def filter_data(inst):
-            inst.data = inst[0:5]
-        self.testInst.custom_attach(filter_data)
-        self.testInst.load(date=self.stime)
-        # starting from no orbit calls next loads first orbit
-        self.testInst.orbits.next()
         # store comparison data
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:376]:
```suggestion

        # Store comparison data
```

File: pysat/tests/test_orbits.py [hunk 17]
```diff
@@ -403,181 +418,74 @@ def filter_data(inst):
         if self.testInst.orbits.num == 1:
             assert all(self.testInst.data == saved_data.data)
 
-        # a recusion issue has been observed in this area
+        # a recursion issue has been observed in this area
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:421]:
```suggestion
        # A recursion issue has been observed in this area.
```

File: pysat/tests/test_orbits.py [hunk 18]
```diff
@@ -403,181 +418,74 @@ def filter_data(inst):
...
         # checking for date to limit reintroduction potential
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:422]:
```suggestion
        # Checking for date to limit reintroduction potential.
```

File: pysat/tests/test_orbits.py [hunk 19]
```diff
@@ -751,13 +658,49 @@ def setup(self):
...
+        return
 
     def teardown(self):
         """Runs after every method to clean up previous testing."""
         del self.testInst, self.stime
+        return
+
+    def test_repeat_orbit_calls_sym_multi_day_0_UT_really_long_time_gap(self):
+        """Test successful orbit calls over a multi-day gap."""
+        self.testInst.load(date=dt.datetime(2009, 1, 19))
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:670]: Set year, month from `self.stime`

File: pysat/tests/test_orbits.py [hunk 20]
```diff
@@ -751,13 +658,49 @@ def setup(self):
...
+            self.testInst.orbits.next()
+        for j in range(45):
+            self.testInst.orbits.prev()
+        assert all(control.data == self.testInst.data)
+        return
 
+    @pytest.mark.parametrize("day", [1, 25])
+    def test_repeat_orbit_calls_cutoffs_with_gaps(self, day):
+        """Test that orbits are selected at same cutoffs when reversed."""
+        self.testInst.load(date=dt.datetime(2009, 1, day))
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:683]: Set year, month from self.stime

File: pysat/tests/test_orbits.py [hunk 21]
```diff
@@ -751,13 +658,49 @@ def setup(self):
...
+        for j in range(20):
+            n_time.append(self.testInst.index[0])
+            self.testInst.orbits.next()
+
+        for j in range(20):
+            self.testInst.orbits.prev()
+            p_time.append(self.testInst.index[0])
+
+        check = np.all(p_time == n_time[::-1])
+        assert all(control.data == self.testInst.data) & check
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:697]: Check one thing at a time for clear assertion errors

File: pysat/tests/test_orbits.py [hunk 22]
```diff
@@ -817,14 +782,18 @@ def setup(self):
...
 
     def teardown(self):
         """Runs after every method to clean up previous testing."""
         del self.testInst, self.stime
+        return
 
 
-class TestOrbitsGappyLongData(TestGeneralOrbitsMLT):
+class TestOrbitsGappyLongData(TestOrbitsGappyData):
+    """Run the gappy orbit tests for orbits defined by Longitude -- pandas"""
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:795]:
```suggestion
    """Run the gappy orbit tests for orbits defined by Longitude -- pandas."""
```

File: pysat/tests/test_orbits.py [hunk 23]
```diff
@@ -834,13 +803,17 @@ def setup(self):
...
 
     def teardown(self):
         """Runs after every method to clean up previous testing."""
         del self.testInst, self.stime
+        return
 
 
-class TestOrbitsGappyLongDataXarray(TestGeneralOrbitsMLT):
+class TestOrbitsGappyLongDataXarray(TestOrbitsGappyData):
+    """Run the gappy orbit tests for orbits defined by Longitude -- xarray"""
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:815]:
```suggestion
    """Run the gappy orbit tests for orbits defined by Longitude -- xarray."""
```

File: pysat/tests/test_orbits.py [hunk 24]
```diff
@@ -849,13 +822,17 @@ def setup(self):
...
+        return
 
     def teardown(self):
         """Runs after every method to clean up previous testing."""
         del self.testInst, self.stime
+        return
+
 
+class TestOrbitsGappyOrbitNumData(TestOrbitsGappyData):
+    """Run the gappy orbit tests for orbits defined by Orbit Number -- pandas"""
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:834]:
```suggestion
    """Run the gappy orbit tests for orbits defined by Orbit Number -- pandas."""
```

File: pysat/tests/test_orbits.py [hunk 25]
```diff
@@ -864,13 +841,17 @@ def setup(self):
...
 
     def teardown(self):
         """Runs after every method to clean up previous testing."""
         del self.testInst, self.stime
+        return
 
 
-class TestOrbitsGappyOrbitNumDataXarray(TestGeneralOrbitsMLT):
+class TestOrbitsGappyOrbitNumDataXarray(TestOrbitsGappyData):
+    """Run the gappy orbit tests for orbits defined by Orbit Number -- xarray"""
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:853]:
```suggestion
    """Run the gappy orbit tests for orbits defined by Orbit Number -- xarray."""
```

File: pysat/tests/test_orbits.py [hunk 26]
```diff
@@ -879,13 +860,17 @@ def setup(self):
...
+        return
 
     def teardown(self):
         """Runs after every method to clean up previous testing."""
         del self.testInst, self.stime
+        return
+
 
+class TestOrbitsGappyOrbitLatData(TestOrbitsGappyData):
+    """Run the gappy orbit tests for orbits defined by Latitude -- pandas"""
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:872]:
```suggestion
    """Run the gappy orbit tests for orbits defined by Latitude -- pandas."""
```

File: pysat/tests/test_orbits.py [hunk 27]
```diff
@@ -895,13 +880,17 @@ def setup(self):
...
 
     def teardown(self):
         """Runs after every method to clean up previous testing."""
         del self.testInst, self.stime
+        return
 
 
-class TestOrbitsGappyOrbitLatDataXarray(TestGeneralOrbitsMLT):
+class TestOrbitsGappyOrbitLatDataXarray(TestOrbitsGappyData):
+    """Run the gappy orbit tests for orbits defined by Latitude -- xarray"""
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:892]:
```suggestion
    """Run the gappy orbit tests for orbits defined by Latitude -- xarray."""
```

**Review by aburrell — COMMENTED** [review]
Block comment grammar needs to be fixed throughout.

File: pysat/tests/test_orbits.py (hunk 13 above)
Comment (jklenzing) [reply pysat/tests/test_orbits.py:206]: I think this is all hard wired.  @rstoneback?

**Comment by jklenzing** [comment]
Last iteration uses `flake8-docstrings` to scan docstrings for consistency.
```
flake8 . --count --select=D,E,F,W
```

File: pysat/tests/test_orbits.py (hunk 7 above)
Comment (jklenzing) [reply pysat/tests/test_orbits.py]: In order to import the instrument test date, I had to move this into the class for `TestOrbitsGappyData`.

File: pysat/tests/test_orbits.py [hunk 28]
```diff
@@ -1,3 +1,5 @@
+"""Test the pysat routines for the orbits class."""
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:1]: Recommend adding the license info to the head of the file (check one of the recent tests I updated).

File: pysat/tests/test_orbits.py [hunk 29]
```diff
@@ -12,47 +14,70 @@
 import pysat
 
 
+def filter_data(inst, times=None):
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:17]:
```suggestion
def filter_data(inst, times):
```

File: pysat/tests/test_orbits.py (hunk 13 above)
Comment (jklenzing) [reply pysat/tests/test_orbits.py:211]: This test and the next one are confusing me.  I get that if no data is loaded it loops around to the last file.  But why is the orbit end 38 minutes here and 11 minutes for the other test?  If we change the test date down the line, are these numbers going to be the same, or will it find different break points?

File: pysat/tests/test_orbits.py [hunk 30]
```diff
@@ -316,82 +349,69 @@ def test_eval_repr(self):
...
 
-        self.testInst.custom_attach(filter_data)
+        self.testInst.custom_attach(truncate_data)
         self.testInst.load(date=self.stime)
         self.testInst.orbits.next()
 
-        # a recusion issue has been observed in this area
-        # checking for date to limit reintroduction potential
+        # A recursion issue has been observed in this area
+        # Checking for date to limit reintroduction potential
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:379]:
```suggestion
        # A recursion issue has been observed in this area.
        # Checking for date to limit reintroduction potential
```

File: pysat/tests/test_orbits.py [hunk 31]
```diff
@@ -403,183 +423,76 @@ def filter_data(inst):
         if self.testInst.orbits.num == 1:
             assert all(self.testInst.data == saved_data.data)
 
-        # a recusion issue has been observed in this area
-        # checking for date to limit reintroduction potential
+        # A recursion issue has been observed in this area
+        # Checking for date to limit reintroduction potential
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:427]:
```suggestion
        # A recursion issue has been observed in this area.
        # Checking for date to limit reintroduction potential
```

File: pysat/tests/test_orbits.py [hunk 32]
```diff
@@ -403,183 +423,76 @@ def filter_data(inst):
...
+        p_time = []
+        for j in range(iterations):
+            n_time.append(self.testInst.index[0])
             self.testInst.orbits.next()
-        for j in range(40):
+        for j in range(iterations):
             self.testInst.orbits.prev()
-        for j in range(20):
-            self.testInst.orbits.next()
+            p_time.append(self.testInst.index[0])
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:445]: This "reversability" code snippit is repeated a lot.  It could probably be an evaluation method in the core class.

**Review by aburrell — COMMENTED** [review]
More comments, though some from my last review are still waiting for input from @rstoneback

File: pysat/tests/test_orbits.py (hunk 3 above)
Comment (jklenzing) [reply pysat/tests/test_orbits.py]: @rstoneback, I've added the check for where the orbit breaks are here, and it is failing for the 45 orbit test on the 19th.  We could drop these added tests for now and add an issue.

Comment (jklenzing) [reply pysat/tests/test_orbits.py]: UPDATE: Since this is a new test, I've dropped the check and added #861 to document

File: pysat/tests/test_orbits.py (hunk 7 above)
Comment (jklenzing) [reply pysat/tests/test_orbits.py]: Updated to use `setup_class` and `teardown_class`

File: pysat/tests/test_orbits.py (hunk 13 above)
Comment (aburrell) [reply pysat/tests/test_orbits.py:284]: Did you create an issue for these fixed date tests?

File: pysat/tests/test_orbits.py [hunk 33]
```diff
@@ -316,82 +436,73 @@ def test_eval_repr(self):
...
-        # store comparison data
+        # Store comparison data
         saved_data = self.testInst.copy()
         self.testInst.load(date=self.stime + dt.timedelta(days=1))
         self.testInst.orbits[0]
         if self.testInst.orbits.num == 1:
-            # equivalence only when only one orbit
-            # some test settings can violate this assumption
+            # Equivalence only when only one orbit
+            # Some test settings can violate this assumption
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:505]:
```suggestion
            # Equivalence occurs only when there is one orbit,
            # some test settings can violate this assumption.
```

**Review by aburrell — COMMENTED** [review]
Looking good, found one small thing.  Want to verify that the hard-coded dates are contained in an issue as well.

**Review by aburrell — APPROVED** [review]

**Comment by jklenzing** [comment]
Added #863 to capture this discussion.

File: pysat/tests/test_orbits.py (hunk 13 above)
Comment (rstoneback) [reply pysat/tests/test_orbits.py:284]: The test date is set on when MLT rolls-over from 24 back to 0, plot attached. 
orbits_mlt

No, if the test date is changed it won't be the same. The correct times can be calculated using the known period and root_date of the 'met' signal.

Comment (rstoneback) [reply pysat/tests/test_orbits.py:284]: The start time check is driven by the fact that it is the first file loaded (last file in following test). The file dates in self.testInst.files.files.index could be used for a more generalized approach.
